### ISSUE TITLE
Infinite loop in JsonRef::resolvePathReferences 

### ISSUE THREAD

**Issue by assertchris** [scrivo/highlight.php#33]
Library version: ^9.13
PHP version: 7.3.1

I'm trying to highlight `javascript` code, and the highlighter gets stuck in an infinite loop while trying to `resolvePathReferences`. Here's a GIF of the diagnosis:

![loop](https://user-images.githubusercontent.com/200609/51088077-29627e80-1764-11e9-8461-ebd2f6aad708.gif)

The code it is compiling for is:

```javascript
const copyToClipboard = function(string) {
    const element = document.createElement('textarea')
    document.body.appendChild(element)

    element.value = string
    element.select()

    // this copies currently selected input text
    document.execCommand('copy')

    document.body.removeChild(element)
}

$('[data-copy-code]').on('click', function(e) {
    copyToClipboard(this.dataset.copyCode)
})
```

...but I don't think the code is the problem here. I suspect a recent-ish change to the javascript syntax has caused this infinite loop behaviour.

Edit: after further investigation, I don't think think a syntax definition change is responsible, since the definition file hasn't changed recently in a way that would cause new behaviour. Still unsure of the cause.

**Comment by allejo** [comment]
I tested running the highlighter on PHP 7.2.10 and it runs just fine. However, it goes into an infinite loop starting with PHP 7.3.0 so something changed between 7.2 and 7.3 that is causing this infinite loop.

**Comment by GKephart** [comment]
I have come across the same issue.
<img width="797" alt="screen shot 2019-01-13 at 9 56 02 pm" src="https://user-images.githubusercontent.com/14102597/51097010-1f4d7980-177e-11e9-9efd-a2c74440d7e6.png">

**Comment by allejo** [comment]
Thanks for the report and fix, @assertchris! Really appreciate it 👍

[v9.13.1.1](https://github.com/scrivo/highlight.php/releases/tag/v9.13.1.1) has been released with your changes from #34 included. I've also added PHP 7.3 and nightly to Travis so that will hopefully help us find these problems sooner.

**Comment by assertchris** [comment]
Thanks for sorting this out so quickly, and sorry for not getting back to you sooner about it!
